zebra-consensus/src/chain.rs

```diff
@@ -79,33 +90,82 @@ where
+        let is_unexpected_high_block = match height {
+            Some(BlockHeight(height))
+                if (height > self.last_block_height.0 + MAX_EXPECTED_BLOCK_GAP) =>
+            {
+                true
+            }
+            Some(height) => {
+                // Update the last height if the block height was expected
+                self.last_block_height = height;
+                false
+            }
+            _ => false,
+        };
+
         async move {
-            // Call a verifier based on the block height and checkpoints
-            //
             // TODO(teor): for post-sapling checkpoint blocks, allow callers
             //             to use BlockVerifier, CheckpointVerifier, or both.
-            match block.coinbase_height() {
+
+            // Call a verifier based on the block height and checkpoints.
+            let hash = match height {
                 Some(height) if (height <= max_checkpoint_height) => {
+                    if info_log {
+                        tracing::info!(?height, "sending block to CheckpointVerifier");
+                    } else {
+                        tracing::debug!(?height, "sending block to CheckpointVerifier");
+                    }
                     checkpoint_verifier
                         .ready_and()
                         .await?
                         .call(block.clone())
                         .await?
                 }
-                Some(_) => {
+                Some(height) => {
+                    // Temporary trace, for identifying early high blocks.
+                    // We think the downloader or sync service should reject these blocks
+                    if is_unexpected_high_block {
+                        tracing::warn!(?height, "unexpected high block");
+                    }
+
+                    if info_log {
+                        tracing::info!(?height, "sending block to BlockVerifier");
+                    } else {
+                        tracing::debug!(?height, "sending block to BlockVerifier");
+                    }
                     block_verifier
                         .ready_and()
                         .await?
                         .call(block.clone())
                         .await?
                 }
-                None => return Err("Invalid block: must have a coinbase height".into()),
+                None => {
+                    tracing::debug!("rejecting block with no coinbase height");
+                    return Err("Invalid block: must have a coinbase height".into());
+                }
             };
 
             // TODO(teor):
             //   - handle chain reorgs
             //   - adjust state_service "unique block height" conditions
 
+            if info_log {
+                tracing::info!(?height, ?hash, "ChainVerifier sent block to state");
+            } else {
+                tracing::debug!(?height, ?hash, "ChainVerifier sent block to state");
+            }
```

Comment:
Similarly to the above, I think this should be a tracing event in the state service, not in the verifier.